data-prepper-plugins/s3-source/src/test/java/org/opensearch/dataprepper/plugins/source/configuration/S3SelectCSVOptionTest.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source.configuration;
+
+import org.junit.jupiter.api.Test;
+import org.opensearch.dataprepper.test.helper.ReflectivelySetField;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class S3SelectCSVOptionTest {
+    @Test
+    void s3SelectOptionsTest() throws NoSuchFieldException, IllegalAccessException {
```

Comment:
I think we can take a slightly different approach to testing this.

1. Create a test which verifies the default values.

```
@Test 
void s3SelectOptions_has_correct_defaults() {
  S3SelectCSVOption csvOption = new S3SelectCSVOption();
  assertThat(csvOption.getFileHeaderInfo(),equalTo(TODO-EXPECTED));
  assertThat(csvOption.getComments(),equalTo(TODO-EXPECTED));
  assertThat(csvOption.getQuiteEscape(),equalTo(TODO-EXPECTED));
}
```


2. Create a test which deserializes from a JSON `ObjectMapper`. Verify the inputs.

```
@Test 
void s3SelectOptions_has_correct_defaults() {
  String knownYaml = "file_header_info: USE\n comments: true";  // TODO: Probably fix this some
  S3SelectCSVOption csvOption = objectMapper.readValue(knownYaml, S3SelectCSVOption.class);
  assertThat(csvOption.getFileHeaderInfo(),equalTo(TODO-EXPECTED));
  assertThat(csvOption.getComments(),equalTo(TODO-EXPECTED));
  assertThat(csvOption.getQuiteEscape(),equalTo(TODO-EXPECTED));
}
```

Reply:
Thanks David for your review, we have incorporated the changes.